skupstina_django/skupstina/utils.py

```diff
@@ -0,0 +1,46 @@
+from django.db.models import Q as QDjango
+from elasticsearch_dsl import Q
+from .documents import ActDocument, act_analyzer
+from .models import Act
+
+base_url = 'http://web.zagreb.hr/'
+
+
+def sql_search(search_term):
+    """Returns result in JSON format"""
+    query_set = Act.objects.filter(
+        QDjango(content__icontains=search_term)
+        | QDjango(subject__icontains=search_term)
+    ).distinct()
+    results = {'matches': [base_url + m.content_url for m in query_set]}
+    return results
+
+
+def elastic_search(search_term):
+    response = act_analyzer.simulate(search_term)
+    tokens = [t.token for t in response.tokens]
+    print(tokens)
+
+    keywords = ['and', 'not']
+
+    if set(keywords).intersection(set(tokens)):
+        q = Q("multi_match", query=tokens[0], fields=['content', 'subject'])
+        for token in tokens:
+            if token in keywords:
+                keyword_position = tokens.index(token)
+                if keyword_position + 1 != len(tokens):
+                    new_query = Q("multi_match", query=tokens[keyword_position+1], fields=['content', 'subject'])
```

Comment:
This is a relatively elegant way to parse the query.  I think there are probably somewhat better ways and parsers are a big topic in computer science (with whole books on the matter – google e.g. "the dragon book"). I think your approach should do for the time being. We should probably make a note in the help article that this won't work if people write two operators one after the other – e.g. "park and not centar", because you would add "not" to the Query. This would probably be a bit of a non-sensical search anyway 🙂

Reply:
I agree. I'm not very happy with the parser at the moment. I've added in another bullet point in the Help to explain this briefly for now.